registry/quilt_server/views.py

```diff
@@ -382,6 +354,97 @@ def wrapper(*args, **kwargs):
         return wrapper
     return innerdec
 
+@app.route('/login', methods=['POST'])
+@api(require_anonymous=True, require_login=False, schema=USERNAME_PASSWORD_SCHEMA)
+@as_json
+def login_post():
+    try:
+        data = request.get_json()
+        username = data.get('username')
+        password = data.get('password')
+        if try_login(username, password):
+            token = issue_token(username)
+        else:
+            return {'error': 'Login attempt failed'}
+
+        db.session.commit()
+        return {'token': token}
+    except Exception:
```

Comment:
You really shouldn't be catching `Exception` here (or elsewhere). Just now, I was getting a 401, "Login failed" because Postgres wasn't running. That should be a 500.